vector/src/main/java/im/vector/app/features/media/domain/usecase/DownloadMediaUseCase.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.media.domain.usecase
+
+import android.content.Context
+import androidx.core.net.toUri
+import dagger.hilt.android.qualifiers.ApplicationContext
+import im.vector.app.core.intent.getMimeTypeFromUri
+import im.vector.app.core.utils.saveMedia
+import im.vector.app.features.notifications.NotificationUtils
+import kotlinx.coroutines.withContext
+import org.matrix.android.sdk.api.session.Session
+import java.io.File
+import javax.inject.Inject
+
+class DownloadMediaUseCase @Inject constructor(
+        @ApplicationContext private val appContext: Context,
+        private val session: Session,
+        private val notificationUtils: NotificationUtils
+) {
+
+    suspend fun execute(input: File): Result<Unit> = withContext(session.coroutineDispatchers.io) {
+        runCatching {
+            saveMedia(
```

Comment:
thinking out loud, I'm wondering if this `usecase` could inline and replace the static `saveMedia` function and if doing so would also help justify the usecase's existence :thinking:   

it would also have the benefit of making any other classes that use `saveMedia` easier to test.

Reply:
Is your suggestion is to move the existing `saveMedia` code into the `useCase`? And then replace all `saveMedia` usage by the `useCase`. If yes, I totally agree since static method are very hard to test. Maybe I can do this in another PR.

Reply:
yep exactly ^^^ a future PR sounds great (potentially a github issue so anyone could pick up the task)

Reply:
I have created the corresponding issue: https://github.com/vector-im/element-android/issues/5374

Reply:
thanks! :100: